pdns/dnsdist-lua-actions.cc

```diff
@@ -553,8 +554,9 @@ DNSAction::Action SpoofAction::operator()(DNSQuestion* dq, std::string* ruleresu
     }
   }
 
+  static std::default_random_engine r;
```

Comment:
Unless `std::default_random_engine::operator()` is thread-safe, I don't think we should make `r` static.

Reply:
Indeed. I do not think we want to re-create the engine object all the time, since it might use a static seed (and it does afaks in at least the clang c++ lib). Maybe a thread local?

Reply:
How about placing it in the `SpoofAction` class? I can write that change if you would like me to :)

Reply:
Right. If that takes care of the multi-threaded access that is preferred. Please do so. There might be caveats I do not know with the `DNSAction` classes :-)

Reply:
On it!

Reply:
For a reason unknown to me Github doesn't allow me to open a pull request against your branch, so please pull https://github.com/PowerDNS/pdns/commit/387221d594adb3a915ff93791e9225a4ecdd8ea7 from https://github.com/rgacogne/pdns/tree/random-engine :)